### ISSUE TITLE
Please put classes into their own namespaces

### ISSUE THREAD

**Issue by mohaaron** [vkhorikov/CSharpFunctionalExtensions#574]
I've been trying to use this library in my projects and keep running into a problem where the Extensions Result is colliding with my own custom Result class library. Could you please put the CSharpFunctionalExtensions classes into area specific name spaces so that I can use them without having a naming collision with my own result class?

**Comment by vkhorikov** [comment]
Yeah, I've thinking of this too, but that's going to be a breaking change, and it's too much of a trouble for people who already use the package.

Why are you using both your own and the library's Results?
